src/Client/Mailer.php

```diff
@@ -0,0 +1,62 @@
+<?php
```

Comment:
Now we're injecting the HTTP client, I think we should make the most of this and add a general `MailerTest` so we can validate the 3 key paths work as intended, for email callouts in general. I think it's enough to check the public fn's return value for the 3 scenarios (we can use a null logger for now as the logs aren't as critical) and this should give us 100% coverage on the new class (rather than near 0)